src/replay/replay_recorder.cpp

```diff
@@ -42,6 +45,12 @@ ReplayRecorder::ReplayRecorder()
 {
     m_complete_replay = false;
     m_incorrect_replay = false;
+
+    // Give margin to store extra events due to higher precision
+    // when a kart's tracked charateristic has suddenly changed
+    // in a non-interpolable way.
+    m_max_frames = (unsigned int)( 1.2f * stk_config->m_replay_max_time
```

Comment:
1.2, hardcoded magic constant

Reply:
The memory to store the replay data is allocated at the start of the recording, for m_max_frames.

While it would be possible to update m_max_frames each time there is a forced update, this would also mean having to resize several arrays each and every time it happens. Though the forced updates don't happen that often, copying huge arrays from one place in memory to another dozens of times would be very wasteful - especially as it is unlikely for m_max_frames to be reached anyway.

If updating dynamically is wasteful, then giving some room at the start for forced updates to fit within the replay max time is more efficient. It doesn't need to be perfect, just to work as long as the player is not trying to break things, by staying entire minutes just before the finish line in the last lap, or spending minutes to catch and burn some nitro, or that sort of thing.

Sure, it would be possible to add an extra config option for this, but is it really worth it ?